Disable wgpu-core documentation as a workaround for #4905.

kpreid opened the pull request:
**Connections**
Proposed workaround for #4905.

**Description**
This enables `cargo doc` to succeed in a reasonable amount of time, as long as the reader isn't looking for documentation for `wgpu-core` itself.

**Testing**
Manually ran `cargo doc` and observed it succeeded, and checked the resulting `wgpu-core` and `wgpu` documentation.

**Checklist**

- [X] Run `cargo fmt`.
- [X] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [X] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

kpreid committed:
Disable wgpu-core documentation as a workaround for #4905.

This enables `cargo doc` to succeed in a reasonable amount of time,
as long as the reader isn't looking for documentation for `wgpu-core`
itself.

cwfitzgerald commented:
Ah, nice workaround, we can also re-enable building docs in CI (currently also commented out)

cwfitzgerald commented:
Would it be possible to add re-enabling CI to this PR?

kpreid commented:
> Would it be possible to add re-enabling CI to this PR?

Besides not building docs, there's a bunch of _stale_ workaround stuff (just search for “4905” in comments), so I'd rather leave that to a separate PR in case of further messes.

kpreid commented:
Oh, wait, I see — if we _just_ enable the docs build now, then _this PR_ gets that aspect tested. Okay.

kpreid committed:
Partial revert "disable rustdoc in CI (#5839)"

This reverts most of the changes in a2fcd72606f83cbb58c1aca2e7e1ad52a11d2067,
but the "document private features" step is still disabled since it
operates only on wgpu-core which is exactly the problem.

cwfitzgerald reviewed (approved):
Amazing, thanks!

kpreid commented:
Fixed changelog merge conflict.

Adjusting the CI script made me think: perhaps the “document wgpu-core anyway” should actually not be a `feature` but a bare `cfg`, so that `--all-features` doesn't enable it. But that means giving users who want the docs the pain of `RUSTFLAGS`.

cwfitzgerald commented:
Hmmm, I think I like the cfg better as wanting this is going to be rare as it's so un-useful

kpreid commented:
> Hmmm, I think I like the cfg better as wanting this is going to be rare as it's so un-useful

Done.

cwfitzgerald reviewed (approved):
Noice
